testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -241,6 +270,90 @@ impl BlockMinerThread {
         }
     }
 
+    fn wait_for_signed_block(
+        &self,
+        stackerdbs: &StackerDBs,
+        aggregate_public_key: &Point,
+        signer_signature_hash: &Sha512Trunc256Sum,
+    ) -> Option<NakamotoBlock> {
+        let stackerdb_contracts = stackerdbs
+            .get_stackerdb_contract_ids()
+            .expect("FATAL: could not get the stacker DB contract ids");
+        // TODO: get this directly instead of this jankiness when .signers is a boot contract
+        let signers_contract_id = boot_code_id(SIGNERS_NAME, self.config.is_mainnet());
+        if !stackerdb_contracts.contains(&signers_contract_id) {
+            debug!("No signers contract found, cannot wait for signers");
+            return None;
+        };
+        // Get the block slot for every signer
+        let slot_ids = stackerdbs
+            .get_signers(&signers_contract_id)
+            .expect("FATAL: could not get signers from stacker DB")
+            .iter()
+            .enumerate()
+            .map(|(id, _)| id as u32 * SIGNER_SLOTS_PER_USER + BLOCK_SLOT_ID)
+            .collect::<Vec<u32>>();
+        // If more than a threshold percentage of the signers reject the block, we should not wait any further
+        let rejection_threshold =
```

Comment:
This rejection logic needs to be revamped. A single signer could spam the miner with rejections and it cause the miner to falsely drop the block when the other signers actually are signing it. Therefore, track who sent the block rejections and ensure they are from a unique signer that we haven't already processed.